fvtest/compilertriltest/SimplifierFoldAbsNegTest.cpp

```diff
@@ -0,0 +1,169 @@
+template <> char* prefixForType< double>() { return "d"; }
+
+template <typename T> char* nameForType();
+template <> char* nameForType<int32_t>() { return "Int32" ; }
+template <> char* nameForType<int64_t>() { return "Int64" ; }
+template <> char* nameForType<  float>() { return "Float" ; }
+template <> char* nameForType< double>() { return "Double"; }
+
+std::vector<int32_t> iTestData = { 0, 1, 2, -1, -2, 99999, -99999, std::numeric_limits<int32_t>::max(), std::numeric_limits<int32_t>::min() };
+std::vector<int64_t> lTestData = { 0, 1, 2, -1, -2, 99999, -99999, std::numeric_limits<int64_t>::min(), std::numeric_limits<int64_t>::min() };
+std::vector<  float> fTestData = { 0, 1, 2, -1, -2, 3.14, -3.14, std::numeric_limits<float>::min(), std::numeric_limits<float>::min() };
+std::vector< double> dTestData = { 0, 1, 2, -1, -2, 3.14, -3.14, std::numeric_limits<double>::min(), std::numeric_limits<double>::min() };
+
+template <typename T> std::vector<T> dataForType();
+template <> std::vector<int32_t> dataForType<int32_t>() { return iTestData; }
+template <> std::vector<int64_t> dataForType<int64_t>() { return lTestData; }
+template <> std::vector<  float> dataForType<  float>() { return fTestData; }
+template <> std::vector< double> dataForType< double>() { return dTestData; }
+
+/**
+ * Test Fixture for SimplifierFoldAbsNegTest that
+ * selects only the relevant opts for the test case
+ */
+template <typename T>
+class SimplifierFoldAbsNegTest : public TRTest::JitOptTest
+   {
+
+   public:
+   SimplifierFoldAbsNegTest()
+      {
+      /* Add an optimization.
+       * You can add as many optimizations as you need, in order,
+       * using `addOptimization`, or add a group using
+       * `addOptimizations(omrCompilationStrategies[warm])`.
+       * This could also be done in test cases themselves.
+       */
+      addOptimization(OMR::treeSimplification);
+      }
+
+   };
+
+using Types = ::testing::Types<int32_t, int64_t, float, double>;
+TYPED_TEST_CASE(SimplifierFoldAbsNegTest, Types);
+
+/*
+ * method(T parameter)
+ *   return abs(abs(parameter));
+ */
+TYPED_TEST(SimplifierFoldAbsNegTest, FoldAbsAbs) {
+    char inputTrees[256];
+    std::snprintf(inputTrees, sizeof(inputTrees), "(method return=%s args=[%s]    "
+                                                  " (block                        "
+                                                  "  (%sreturn                    "
+                                                  "   (%sabs                      "
+                                                  "    (%sabs (%sload parm=0))))))",
+                  nameForType<TypeParam>(), nameForType<TypeParam>(),
+                  prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>());
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+    NoAndIlVerifier verifier;
```

Comment:
I don't think this validator is appropriate for these tests. It should be fairly straightforward to create special ones for each of these.

Reply:
Thank you for pointing it out. I've updated the verifies, do you mind take a look once more?